### PR TITLE
8288474: Move EventContinuationFreezeOld from try_freeze_fast to freeze_slow

### PR DESCRIPTION
**Pull request by coleenp**
This moves the event ContinuationFreezeOld to freeze_slow so it's reported on the different paths to freeze_slow, and renames the events Continuation*Old/Continuation*Slow and Continuation*Young/Continuation*Fast to reflect that they're on the fast/slow paths.  I turned on CONT_JFR and fixed some bugs and turned it back off again.  Some output:

$ jfr summary hotspot-pid-31671-id-1-2022_09_13_19_42_01.jfr | grep Continuation
 jdk.ContinuationThaw                    12193        211929
 jdk.ContinuationThawFast                10139        165420
 jdk.ContinuationFreeze                   3196         57135
 jdk.ContinuationFreezeFast               2788         46397
 jdk.ContinuationThawSlow                 2054         28543
 jdk.ContinuationFreezeSlow                408          5688

Tested locally and with tier7 which seems to run JFR tests.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8288474](https://bugs.openjdk.org/browse/JDK-8288474): Move EventContinuationFreezeOld from try_freeze_fast to freeze_slow

### Reviewers
 * [Ron Pressler](https://openjdk.org/census#rpressler) (@pron - Committer)
 * [Patricio Chilano Mateo](https://openjdk.org/census#pchilanomate) (@pchilano - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10254/head:pull/10254` \
`$ git checkout pull/10254`

Update a local copy of the PR: \
`$ git checkout pull/10254` \
`$ git pull https://git.openjdk.org/jdk pull/10254/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10254`

View PR using the GUI difftool: \
`$ git pr show -t 10254`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10254.diff">https://git.openjdk.org/jdk/pull/10254.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/runtime/continuationFreezeThaw.cpp [hunk 1]
```diff
@@ -724,6 +718,14 @@ NOINLINE freeze_result FreezeBase::freeze_slow() {
   log_develop_trace(continuations)("freeze_slow  #" INTPTR_FORMAT, _cont.hash());
   assert(_thread->thread_state() == _thread_in_vm || _thread->thread_state() == _thread_blocked, "");
 
+#if CONT_JFR
+  EventContinuationFreezeSlow e;
+  if (e.should_commit()) {
```
Comment (pron) [inline src/hotspot/share/runtime/continuationFreezeThaw.cpp:723]: Committing the event should happen at the end of the method to measure the time.

File: src/hotspot/share/runtime/continuationFreezeThaw.cpp [hunk 2]
```diff
@@ -1848,12 +1850,13 @@ NOINLINE intptr_t* ThawBase::thaw_slow(stackChunkOop chunk, bool return_barrier)
     chunk->print_on(true, &ls);
   }
 
-  // Does this need ifdef JFR around it? Or can we remove all the conditional JFR inclusions (better)?
-  EventContinuationThawOld e;
+#if CONT_JFR
+  EventContinuationThawSlow e;
   if (e.should_commit()) {
```
Comment (pron) [inline src/hotspot/share/runtime/continuationFreezeThaw.cpp:1855]: Committing the event should be done at the end of the method.

File: src/hotspot/share/runtime/continuationFreezeThaw.cpp (hunk 1 above)
Comment (coleenp) [reply src/hotspot/share/runtime/continuationFreezeThaw.cpp:723]: That makes sense but the time recorded for these is startTime:
jdk.ContinuationThawSlow {
  startTime = 11:54:23.301 (2022-09-14)
  id = 4260565248
  eventThread = "MainThread" (javaThreadId = 36)
}

**Review by pron — APPROVED** [review]

**Review by pchilano — APPROVED** [review]
Looks good to me!

**Comment by coleenp** [comment]
Thanks Ron and Patricio!
/integrate

**Pull request closed without merging**